### PR TITLE
GH-447: As a developer I want that projects are not built if their manifest files contain error markers

### PR DESCRIPTION
**Pull request by mmews-n4**
~~see #429~~ (see [comment below](https://github.com/eclipse/n4js/pull/431#issuecomment-356852571)),
see #447

### CODE REVIEW

File: plugins/org.eclipse.n4js/src/org/eclipse/n4js/utils/ResourceType.java [hunk 1]
```diff
@@ -70,6 +71,14 @@ public static ResourceType getResourceType(Resource resource) {
 		return getResourceType(resource.getURI());
 	}
 
+	// todo
```
Comment (mor-n4) [inline plugins/org.eclipse.n4js/src/org/eclipse/n4js/utils/ResourceType.java]: there are several of these still in the code

File: tests/org.eclipse.n4js.ui.tests/src/org/eclipse/n4js/tests/externalPackages/IDEBUG_855_PluginUITest.java [hunk 2]
```diff
@@ -68,6 +69,9 @@
 	@Inject
 	private ExternalLibraryPreferenceStore externalLibraryPreferenceStore;
 
+	@Inject
+	private N4JSBuilderPreferenceAccess prefAccess;
```
Comment (mor-n4) [inline tests/org.eclipse.n4js.ui.tests/src/org/eclipse/n4js/tests/externalPackages/IDEBUG_855_PluginUITest.java]: as discussed: I'd also vote for not making the new behavior configurable

File: plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/building/N4JSBuildTypeTrackingBuilder.java [hunk 3]
```diff
@@ -74,7 +82,25 @@ protected void doBuild(ToBeBuilt toBeBuilt, IProgressMonitor monitor, BuildType
...
+				// Only build the project if the manifest file is error free OR if setting is disabled
+				super.doBuild(toBeBuilt, monitorPJ, type);
+			} else {
+				String completeLogMsg = "\n";
+				completeLogMsg += "Errors in Manifest. Project is not built.\n";
+				completeLogMsg += "Manifest errors:\n";
+				for (String errMsg : errMsgsInManifest) {
+					completeLogMsg += errMsg + "\n";
+				}
+				LOGGER.warn(completeLogMsg);
```
Comment (mor-n4) [inline plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/building/N4JSBuildTypeTrackingBuilder.java]: not sure if this should be logged as a warning.
From the perspective of IDE usage, having a project with errors in the manifest is an ordinary use case (e.g. happens all the time while editing/refactoring things)

**Comment by mor-n4** [comment]
While experimenting with this, I got:
```
java.lang.NullPointerException
	at org.eclipse.n4js.ui.building.N4JSBuildTypeTrackingBuilder.getErrorMsgsInManifest(N4JSBuildTypeTrackingBuilder.java:153)
	at org.eclipse.n4js.ui.building.N4JSBuildTypeTrackingBuilder.doBuild(N4JSBuildTypeTrackingBuilder.java:90)
	at org.eclipse.xtext.builder.impl.XtextBuilder.incrementalBuild(XtextBuilder.java:267)
	at org.eclipse.n4js.ui.building.N4JSBuildTypeTrackingBuilder.incrementalBuild(N4JSBuildTypeTrackingBuilder.java:62)
	at org.eclipse.xtext.builder.impl.XtextBuilder.build(XtextBuilder.java:161)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:735)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:206)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:246)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:301)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:304)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:360)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:383)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:144)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:235)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:55)
```

**Comment by mmews-n4** [comment]
Thank you for your reviews and the discussion yesterday. As we agreed, I will remove the implementation regarding the change in the build procedure. Instead, only the fixes to the test cases will be merged, eventually, in this PR. All other changes are postponed to a follow-up userstory.

**Comment by mmews-n4** [comment]
No, I wasn't finished yet. The change to ResourceType was intended to stay since I thought that the two new methods could be handy/convenience. However, removing them is also fine.

**Comment by mmews-n4** [comment]
@yoosiba @mor-n4 This should be a mergable version now. Please have a second look.

**Pull request merged**